metric_tracking/__init__.py

```diff
@@ -0,0 +1,241 @@
+import logging
+import storage
+import mlflow
+
+def do_tracking():
+    available_dates = list(storage.available().keys())
+    available_dates.sort()
+
+    for date_a, date_b in zip(available_dates[:-1], available_dates[1:]):
+        mlflow.start_run()
+        mlflow.log_param("Date A", date_a)
+        mlflow.log_param("Date B", date_b)
+        calculate_score(date_a, date_b)
+        mlflow.end_run()
+
+def calculate_score(date_a, date_b):
+    cluster_a = storage.read(date_a)
+    cluster_b = storage.read(date_b)
+
+    inverse_cluster_a = invert_cluster(cluster_a)
+    inverse_cluster_b = invert_cluster(cluster_b)
+
+    logging.info("calculate score for dates %s - %s" % (date_a, date_b))
+    for cluster_id in inverse_cluster_a.keys():
+        systems_a = frozenset(inverse_cluster_a[cluster_id])
+        systems_b = frozenset(inverse_cluster_b[cluster_id])
+        # So now if you have 100 system ids in first and 160 in second
+        # (100 are the same and we have 60 new), you'll count the score as
+        # 100 / (100+160/2) = 100 / 130 = 0.76.
+        # While it should be 1, if the 100 systems were clustered the same.
+        # We can't say anything about the 60 new systems (unless we do other analysis of them),
+        # so those needs to be ignored.
+        # TODO: only look at systems present in both runs
```

Comment:
apart from https://github.com/RedHatInsights/aicoe-insights-clustering/pull/13#discussion_r222564392

Another thing that will definitely hit the stability is using wrong amount of clusters. E.g. we should set them by doing elbow method, this has a nice example https://mubaris.com/2017/10/01/kmeans-clustering-in-python/#choosing-the-value-of-k, it also shows that incorrect amount cluster will lead to unstable and wrong results.

Reply:
hm, I see you have all of these in your Cluster Validation ideas document